src/scripts/main.js

```diff
@@ -1,5 +1,26 @@
 'use strict';
 
+const spider = document.querySelector('.spider');
+const wall = document.querySelector('.wall');
+
 document.addEventListener('click', (e) => {
-  // write code here
+  const field = e.target.closest('.wall');
+
+  if (!field) {
+    return;
+  }
+
+  const wallRect = wall.getBoundingClientRect();
+  const spiderRect = spider.getBoundingClientRect();
+  const widthSpider = spider.offsetWidth;
+  const border = wall.clientTop;
+
+  let x = e.clientX - widthSpider / 2 - border - wallRect.left;
+  let y = e.clientY - widthSpider / 2 - border - wallRect.top;
+
+  x = Math.max(0, Math.min(x, wall.clientWidth - spiderRect.width));
+  y = Math.max(0, Math.min(y, wall.clientHeight - spiderRect.height));
```

Comment:
These lines are ensuring that the spider doesn't move outside of the wall. This is a critical part of the functionality and should be separated into its own function for clarity and potential reuse.